File: game.py
import numpy as np


class PokerGame:
    def __init__(self):
        self.deck = self.create_deck()
        self.community_cards = []
        self.player_hands = []

    def create_deck(self):
        suits = ['Hearts', 'Diamonds', 'Clubs', 'Spades']
        ranks = range(2, 15)  # e.g. Jack(11), Queen(12), etc..
        return [(rank, suit) for suit in suits for rank in ranks]

    def deal_cards(self, num_cards):
        return [self.deck.pop(np.random.randint(len(self.deck))) for _ in range(num_cards)]

    def start_round(self):
        self.deck = self.create_deck()
        self.community_cards = self.deal_cards(5)
        self.player_hands = [self.deal_cards(5)
                             for _ in range(5)]  # Five players

    def show_game_state(self):
        print("Community Cards:", self.community_cards)
        for i, hand in enumerate(self.player_hands):
            print(f"Player {i+1} Hand: {hand}")

File: game.jl
module PokerGame

export start_round, show_game_state

function create_deck()
    suits = ["Hearts", "Diamonds", "Clubs", "Spades"]
    ranks = 2:14  # e.g. Jack(11), Queen(12), King(13), Ace(14)
    return [(rank, suit) for suit in suits for rank in ranks]
end

function deal_cards!(deck, num_cards)
    hand = []
    for _ in 1:num_cards
        card = pop!(deck, rand(1:length(deck)))
        push!(hand, card)
    end
    return hand
end

mutable struct GameState
    deck::Array
    community_cards::Array
    player_hands::Array
end

function start_round(game::GameState)
    game.deck = create_deck()
    game.community_cards = deal_cards!(game.deck, 5)
    game.player_hands = [deal_cards!(game.deck, 5) for _ in 1:5]  # Five player game
end

function show_game_state(game::GameState)
    println("Community Cards: ", game.community_cards)
    for (i, hand) in enumerate(game.player_hands)
        println("Player $i Hand: ", hand)
    end
end

end
